File: 백준/Silver/10773. 제로/제로.py
"""
"""
import sys
                                  
K = int(sys.stdin.readline().strip())   # 케이스 갯수 받아주고

stack = []                              # 스택 선언 후
for _ in range(K):                      # K번 만큼 반복한다.
  money = int(sys.stdin.readline().strip()) # 돈 받아서
  
  if money != 0:                        # 0아닌겨
    stack.append(money)                 # 넣고
  
  if money == 0:                        # 0인겨
    stack.pop()                         # 빼고

print(sum(stack))                       # 합출력

File: 백준/Silver/10773. 제로/제로.js
// /dev/stdin
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map((x) => x.replace("\r", ""));

const N = Number(input[0]);

const stack = [];

for (let i = 1; i <= N; i++) {
  if (input[i] == 0) {
    stack.pop();
  } else {
    stack.push(Number(input[i]));
  }
}

console.log(stack.reduce((acc, cur) => acc + cur, 0));
